### ISSUE TITLE
stack also works

### ISSUE THREAD

**Issue by juhp** [srid/ema-docs#3]
I know nix is getting attention but not everyone (including me) uses it...
and in fact stack also works fine I think for running this project.

Would you accept a PR to add stack.yaml and an explanation for how to run with stack?
It should be pretty simple.

**Comment by srid** [comment]
Absolutely. Nix is only a "nice to have", it is not required to use this project. 

If you are using it outside of Nix, however, you should remember to run the project via ghcid yourself (otherwise hot reload on code change wouldn't work, of course). The `bin/run` script does this, via nix.

> Would you accept a PR to add stack.yaml and an explanation for how to run with stack?

Informing users how to use this template with Stack or even plain Cabal is a great idea. The problem with adding stack.yaml to the repo however is that it can soon become out of date with no maintainer to fix it (I don't use stack myself). We could perhaps add instructions to README, or even better - add a new page under https://ema.srid.ca/guide ?

**Comment by juhp** [comment]
> Absolutely. Nix is only a "nice to have", it is not required to use this project.

Okay I will open a PR then later.

> If you are using it outside of Nix, however, you should remember to run the project via ghcid yourself (otherwise hot reload on code change wouldn't work, of course). The `bin/run` script does this, via nix.

(Not sure if I am doing something wrong but I couldn't seem to get content hot reloading to work without restarting but I will test more on that.)

> > Would you accept a PR to add stack.yaml and an explanation for how to run with stack?
> 
> Informing users how to use this template with Stack or even plain Cabal is a great idea. The problem with adding stack.yaml to the repo however is that it can soon become out of date with no maintainer to fix it (I don't use stack myself).

The problem is stack requires stack.yaml file to do anything (even to fail;) - IMO even a broken (slightly outdated) stack.yaml is still better than none. But I hear you - well it would not be hard to add stack to the CI even - you could just leave it users to update it as needed. Also not a bad idea to add the deps/packages to Stackage too. :-)

**Comment by simonmichael** [comment]
Here's one that worked yesterday, FWIW:
```
resolver: lts-18.5
packages:
- .

allow-newer: true

extra-deps:
- aeson-1.5.6.0
- aeson-extra-0.5
- dlist-0.8.0.8
- lvar-0.1.0.0
- map-syntax-0.3
- monad-logger-extras-0.1.1.1
- shower-0.2.0.2
- th-abstraction-0.4.2.0
- xmlhtml-0.2.5.2
- git: https://github.com/srid/ema
  commit: 82ce6681d876e961a84eac5650b06eb86535521c
- git: https://github.com/srid/pandoc-link-context
  commit: 89a5bebdf765d331d32a34d75abe040816ad5d5c
- git: https://github.com/srid/tagtree
  commit: 667c5a5073eddb8ddd875726b1fb6c41e09f0591
- git: https://github.com/srid/heist
  commit: a8be6f93656f87199e03b9dfb9979dc061614108
```

**Comment by simonmichael** [comment]
(Obviously allow-newer:true isn't ideal, some work is needed on the deps to get rid of that.

**Comment by srid** [comment]
@simonmichael I believe this stack.yaml is for https://github.com/srid/emanote-template

(ema is the library that emanote uses)

But yes, I do plan to release ema, pandoc-link-context and tagtree to Hackage. We should probably put the version constraints on the .cabal file too. Then a simple 'cabal install' should work - but only after https://github.com/snapframework/heist/issues/129 is done.
